tss/keygen.go

```diff
@@ -1,29 +1,33 @@
 package tss
 
 import (
-	"errors"
 	"time"
 
 	"github.com/bnb-chain/tss-lib/crypto"
 	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
 
 	"github.com/zeta-chain/go-tss/blame"
 	"github.com/zeta-chain/go-tss/common"
 	"github.com/zeta-chain/go-tss/conversion"
 	"github.com/zeta-chain/go-tss/keygen"
 	"github.com/zeta-chain/go-tss/keygen/ecdsa"
 	"github.com/zeta-chain/go-tss/keygen/eddsa"
+	"github.com/zeta-chain/go-tss/logs"
 	"github.com/zeta-chain/go-tss/messages"
-	"github.com/zeta-chain/go-tss/p2p"
 )
 
 func (t *Server) Keygen(req keygen.Request) (keygen.Response, error) {
 	t.tssKeyGenLocker.Lock()
 	defer t.tssKeyGenLocker.Unlock()
-	status := common.Success
-	msgID, err := t.requestToMsgID(req)
+
+	if req.Version != messages.VersionJoinPartyWithLeader {
+		return keygen.Response{}, errors.Errorf("invalid version %q", req.Version)
+	}
+
+	msgID, err := req.MsgID()
 	if err != nil {
-		return keygen.Response{}, err
+		return keygen.Response{}, errors.Wrap(err, "unable to get message id")
```

Comment:
`fmt.Errorf`

Reply:
I think wrap is cleaner, allow to unwrap the error

Reply:
> I think wrap is cleaner, allow to unwrap the error

`fmt.Errorf` can be unwrapped with `errors.Is` and `errors.As`